### PR TITLE
Refactor geoip-fixture to remove Docker requirement

### PR DESCRIPTION
**Pull request by mark-vieira**
This removes the last remaining use of test-clusters gradle plugin for full cluster restart tests, as well as eliminates the need to use Docker for the geoip-fixture. We instead run it in process in the test jvm.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@joegallo Perhaps you could help me track down why I'm getting a failure in the `FullClusterRestartTest`.

https://gradle-enterprise.elastic.co/s/vsqkmk4hlm5nk/tests/:modules:ingest-geoip:qa:full-cluster-restart:v7.17.9%23bwcTest/org.elasticsearch.ingest.geoip.FullClusterRestartIT/testGeoIpSystemFeaturesMigration%20%7Bcluster=UPGRADED%7D?top-execution=1

I'm seeing the following in the cluster logs after restart and suspect this might have something to do with it?

```
[2023-02-03T02:53:46,921][ERROR][o.e.i.g.GeoIpDownloader  ] [test-cluster-1] exception during geoip databases update org.elasticsearch.ElasticsearchException: not all primary shards of [.geoip_databases] index are active
	at org.elasticsearch.ingest.geoip@8.7.0-SNAPSHOT/org.elasticsearch.ingest.geoip.GeoIpDownloader.updateDatabases(GeoIpDownloader.java:131)
	at org.elasticsearch.ingest.geoip@8.7.0-SNAPSHOT/org.elasticsearch.ingest.geoip.GeoIpDownloader.runDownloader(GeoIpDownloader.java:279)
	at org.elasticsearch.ingest.geoip@8.7.0-SNAPSHOT/org.elasticsearch.ingest.geoip.GeoIpDownloaderTaskExecutor.nodeOperation(GeoIpDownloaderTaskExecutor.java:157)
	at org.elasticsearch.ingest.geoip@8.7.0-SNAPSHOT/org.elasticsearch.ingest.geoip.GeoIpDownloaderTaskExecutor.nodeOperation(GeoIpDownloaderTaskExecutor.java:56)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.persistent.NodePersistentTasksExecutor$1.doRun(NodePersistentTasksExecutor.java:42)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:958)
	at org.elasticsearch.server@8.7.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by joegallo** [comment]
I got you, @mark-vieira. It was 2e8a9643bffe2da28c171da2ea774f56260d38b5.

**Comment by mark-vieira** [comment]
> I got you, @mark-vieira. It was https://github.com/elastic/elasticsearch/commit/2e8a9643bffe2da28c171da2ea774f56260d38b5.

🤦 Thanks.

File: modules/ingest-geoip/src/yamlRestTest/java/org/elasticsearch/ingest/geoip/IngestGeoIpClientYamlTestSuiteIT.java [hunk 1]
```diff
@@ -31,6 +38,25 @@
 
 public class IngestGeoIpClientYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
 
+    private static final boolean useFixture = Booleans.parseBoolean(System.getProperty("geoip_use_service", "false")) == false;
```
Comment (rjernst) [inline modules/ingest-geoip/src/yamlRestTest/java/org/elasticsearch/ingest/geoip/IngestGeoIpClientYamlTestSuiteIT.java:41]: Why does this use `Booleans.parseBoolean`, but the other test FullClusterRestartIT uses `Boolean.getBoolean`?

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/ingest-geoip/src/yamlRestTest/java/org/elasticsearch/ingest/geoip/IngestGeoIpClientYamlTestSuiteIT.java (hunk 1 above)
Comment (mark-vieira) [reply modules/ingest-geoip/src/yamlRestTest/java/org/elasticsearch/ingest/geoip/IngestGeoIpClientYamlTestSuiteIT.java:41]: It sounds like we aren't applying the same forbidden apis rules to all source sets the same then. I only got a failure in this case.

**Pull request merged**